dashboard/app/models/teacher_score.rb

```diff
@@ -54,4 +54,40 @@ def self.score_level_for_student(
       score: score
     )
   end
+
+  # When a teacher marks a stage complete for a section, that stage is marked
```

Comment:
@davidsbailey here's the part that seems dicey. I'm not sure whether to check if _any_ student in the section has completed the lesson or if _all_ students in the section completed the lesson. Or if we should be using the X% of students have teacher scores on  %Y of levels logic to mirror plugged

Reply:
yes, we should use the X% of students have teacher scores on %Y of levels logic to mirror plugged! :-)

Reply:
I think this logic should be more general though. when a teacher marks a level with a score, that's going to need to show up everywhere we show progress, right? so, the teacher score needs to get merged into whatever progress data we pull from the DB. and then you may be able to use existing logic for X% / Y% and treat unplugged levels the same as plugged levels.

Reply:
that way we'll basically be implementing "progress resetting" too. so when its time to build that feature, we'll only really need to add a new UI for a teacher to add teacher scores of 0 in the right places.